src/messageprocessor.ts

```diff
@@ -56,6 +33,24 @@ export class MessageProcessorMatrixResult {
     public msgtype: string;
 }
 
+interface IUserNode {
```

Comment:
I feel like we could just call this IDiscordMsgNode and have just IEnojiNode inherit it, full OOP style.